src/accountsdb/index.zig

```diff
@@ -39,7 +38,32 @@ pub const AccountReferenceHead = RwMux(struct {
 
         return .{ rooted_ref_count, ref_slot_max };
     }
-});
+
+    pub const PtrToFieldThatIsPtrToRef = union(enum) {
+        null,
+        head,
+        inner: *?*AccountRef,
+    };
+    /// Returns a pointer to the field which is a pointer to the
+    /// account reference pointer with a field `.slot` == `slot`.
+    /// Returns `.null` if no account reference has said slot value.
+    /// Returns `.head` if `head_ref.ref_ptr.slot == slot`.
+    /// Returns `.inner = ptr` if `ptr.*.?.*.slot == slot`.
+    pub inline fn getPtrToFieldThatIsPtrToRefWithSlot(
+        head_ref: *const AccountReferenceHead,
+        slot: Slot,
+    ) PtrToFieldThatIsPtrToRef {
+        if (head_ref.ref_ptr.slot == slot) return .head;
+        var curr_ref_ptr_ptr: *?*AccountRef = &head_ref.ref_ptr.next_ptr;
```

Comment:
same with this naming

Reply:
0745b4b6290b777c764788678cf9834d20df8a7e